cmd/sonobuoy/app/gen_test.go

```diff
@@ -244,44 +257,105 @@ func TestResolveConfig(t *testing.T) {
+					{SonobuoyConfig: manifest.SonobuoyConfig{Driver: "Job", PluginName: "plugin3", SkipCleanup: false, ResultFormat: "raw"}, Spec: manifest.Container{Container: v1.Container{Name: "plugin", Image: "foo/bar:v3", Command: []string{"./run.sh"}, VolumeMounts: []v1.VolumeMount{{ReadOnly: false, Name: "results", MountPath: plugin.ResultsDir}}}}},
+				},
+				DynamicPlugins: []string{"e2e"},
+			},
 		},
 	}
 
 	for _, tc := range tcs {
 		t.Run(tc.name, func(t *testing.T) {
-			// Simulate parsing of input via CLI. Making this optional to avoid complicating
-			// setup for other tests which just explicitly set the values.
-			if len(tc.cliInput) > 0 {
-				fs := GenFlagSet(tc.input, EnabledRBACMode)
-				if err := fs.Parse(strings.Split(tc.cliInput, " ")); err != nil {
-					t.Fatalf("Failed to parse CLI input %q: %v", tc.cliInput, err)
+			// Write and load a config file
+			if len(tc.configFileContents) > 0 {
+				tmpFile, err := ioutil.TempFile("", "")
+				if err != nil {
+					t.Fatalf("Failed to create temp file for test: %v", err)
+				}
+				defer os.Remove(tmpFile.Name())
+				err = ioutil.WriteFile(tmpFile.Name(), []byte(tc.configFileContents), 0666)
+				if err != nil {
+					t.Fatalf("Failed to write temp config file for test: %v", err)
 				}
+				tc.input = strings.Replace(tc.input, dynamicConfigFileName, tmpFile.Name(), -1)
 			}
 
-			conf := tc.input.resolveConfig()
+			// Simulate parsing of input via CLI.
+			gflagset := &genFlags{}
+			fs := GenFlagSet(gflagset, EnabledRBACMode)
+			if err := fs.Parse(strings.Split(tc.input, " ")); err != nil {
+				t.Fatalf("Failed to parse CLI input %q: %v", tc.input, err)
+			}
+			genConfig, err := gflagset.Config()
+			if err != nil {
+				t.Fatalf("Failed to generate GenConfig: %v", err)
+			}
+
+			if genConfig.Config.Namespace != tc.expected.Namespace {
+				t.Errorf("Expected namespace %v but got %v", tc.expected.Namespace, genConfig.Config.Namespace)
+			}
+
+			if genConfig.Config.WorkerImage != tc.expected.WorkerImage {
+				t.Errorf("Expected worker image %v but got %v", tc.expected.WorkerImage, genConfig.Config.WorkerImage)
+			}
 
-			if conf.Namespace != tc.expected.Namespace {
-				t.Errorf("Expected namespace %v but got %v", tc.expected.Namespace, conf.Namespace)
+			if genConfig.Config.ImagePullPolicy != tc.expected.ImagePullPolicy {
+				t.Errorf("Expected image pull policy %v but got %v", tc.expected.ImagePullPolicy, genConfig.Config.ImagePullPolicy)
 			}
 
-			if conf.WorkerImage != tc.expected.WorkerImage {
-				t.Errorf("Expected worker image %v but got %v", tc.expected.WorkerImage, conf.WorkerImage)
+			if genConfig.Config.Aggregation.TimeoutSeconds != tc.expected.Aggregation.TimeoutSeconds {
```

Comment:
Now that we are generating the 'higher level' genConfig object, these checks just needed tweaked to look at the same value.